packages/core/src/templates/contract/multisigproxy.handlebars

```diff
@@ -2,25 +2,49 @@
  * This file is auto-generated using abi-gen. Any changes will be reverted
  */
 
+ {{!-- Some contracts have inputs like `_value` which run afoul of linter, so: --}}
+ /* tslint:disable:variable-name */
+
 import { isDefined, bindNestedAll } from "@joincivil/utils";
 import BigNumber from "bignumber.js";
 
 import { createTwoStepSimple } from "../../utils/contracts";
 import { BaseMultisigProxy, MultisigProxyTransaction } from "../../multisig/basemultisigproxy";
-import { OwnableContract } from "../../interfaces/ownable";
+import { Contract } from "../../interfaces/contract";
 import { EthApi } from "../../../utils/ethapi";
 import { TxData } from "../../../types";
 
 // TODO(ritave): Abi-Gen doesn't support lowercase contractName so we can't
 //               import the Contract wrappers, we're using generics instead
 //               This means calls to instance are untyped inside are using (this: any) type
 export class {{contractName}}MultisigProxy extends BaseMultisigProxy {
+  /**
+   * Creates a multisig wrapper around a contract
+   * @param ethApi Web3 API
+   * @param instance Contract we are wrapping
+   * @param multisigAddress (optional) Address of multisig wallet - if not supplied, will check if contract is owned by a multisig wallet and use that instead
+   */
   public static async create(
     ethApi: EthApi,
-    instance: OwnableContract,
+    instance: Contract,
+    multisigAddress?: string,
   ): Promise<{{contractName}}MultisigProxy> {
     const proxy = new {{contractName}}MultisigProxy(ethApi, instance);
-    await proxy.resolveMultisig();
+    await proxy.resolveMultisig(multisigAddress);
+    return proxy;
+  }
+
+  /**
+   * Creates a wrapper around a contract with identical API to multisig wrapper, but all transactions are passed directly to contract. Essentially just a helper for when we know multisig isn't needed and want a synchronous call.
+   * @param ethApi Web3 API
+   * @param instance Contract we are wrapping
+   */
+  public static createNonMultisig(
```

Comment:
Just copy-pasting my comment about this since the comment thread is hidden now because the code it was on is gone.

The reason I made function is to allow a synchronous way of instantiating TCR/token when multisig is not needed, which is 95% of the time. Existing `tcrSingletonTrusted` is synchronous but multisig version must be async, and there are 100+ synchronous pathways that lead to `tcrSingletonTrusted` (e.g. every `getCivil()` call in dapp) , and updating them all to be async all the way through chain of calls seemed crazy.

We need this also so we can always rely on a multisig proxy being present with the same API, even if it's not actually using multisig. On the one hand this feels weird, on the other hand it prevents dozens of `if (this.multisigProxy) { ... } else { .... }` branches.